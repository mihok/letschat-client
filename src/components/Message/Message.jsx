import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import './Message.css';

const CHAT_OPERATOR = 'OPERATOR';

export const MessageComponent = (props) => {
  const { chatStyle, author, timestamp, key } = props;
  const chatStyleName = chatStyle.toLowerCase();
  const content = props.content.map((message, index) => <li key={index}>{message}</li>);

  let message = (
    <div>
      <ul className={`message-userContent-${chatStyleName}`}>
        {content}
      </ul>
    </div>
  );

  if (author === CHAT_OPERATOR) {
    message = (
      <div>
        <div className={`message-operatorPicture-${chatStyleName}`}>
          <img alt="Operator" className={`message-operatorPictureImage-${chatStyleName}`} src="http://placehold.it/40x40/" />
        </div>
        <ul className={`message-operatorContent-${chatStyleName}`}>
          {content}
        </ul>
      </div>
    );
  }

  return (
    <li id={`message_${key}`} className={`message-message-${chatStyleName}`} style={{ clear: 'both' }}>
      {message}
      <span className={`message-message-timestamp-${chatStyleName}`}>{timestamp}</span>
    </li>
  );
};

MessageComponent.propTypes = {
  chatStyle: PropTypes.string,
  timestamp: PropTypes.number,
  key: PropTypes.string,
  author: PropTypes.string,
  content: PropTypes.arrayOf(
    PropTypes.string,
  ),
};

const mapStateToProps = state => ({
  chatStyle: state.ui.style,
});

const mapDispatchToProps = dispatch => ({
  dispatch,
});


const Message = connect(
  mapStateToProps,
  mapDispatchToProps,
)(MessageComponent);

export default Message;
