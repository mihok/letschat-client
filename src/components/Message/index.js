import { h } from 'preact';
import PropTypes from 'prop-types';

import { applyTheme } from '../ThemeProvider';

import './styles.css';

const Message = props => {
  // our message's content
  const content = props.content.map((msg, i) =>
    <li key={i}>
      {msg}
    </li>
  );

  // our default message is a client message
  let message = (
    <div className={`Message__operatorWrapper--${props.theme}`}>
      <ul className={`Message__operator--${props.theme}`}>
        {content}
      </ul>

      <img
        alt="Operator"
        className={`Message__avatar--${props.theme}`}
        src="http://placehold.it/40x40/"
      />
    </div>
  );

  // Ff the iterated message is an operator; override `message`
  if (props.type.indexOf('client') >= 0) {
    message = (
      <ul className={`Message__client--${props.theme}`}>
        {content}
      </ul>
    );
  }

  // Incoming props, mesage.content is an array.
  return (
    <li className={`Message__box--${props.theme}`}>
      {message}
    </li>
  );
};

Message.propTypes = {
  theme: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  content: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default applyTheme(Message);
