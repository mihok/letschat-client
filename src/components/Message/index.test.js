import render from 'preact-render-to-string';
import { h } from 'preact';
import Message from '.';

describe('Message component', () => {
  it('should render ', () => {
    const props = {
      content: [], // message content
      type: "client-tihodijakl"
    };

    const tree = render(<Message type={props.type} content={props.content} />);
    expect(tree).toMatchSnapshot();
  });
});
