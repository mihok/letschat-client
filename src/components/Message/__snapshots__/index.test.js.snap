// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component should render  1`] = `"<li class=\\"Message__box--undefined\\"><ul class=\\"Message__client--undefined\\"></ul></li>"`;
